File: pset6/caesar/caesar.py
from cs50 import get_string
import sys

if not len(sys.argv) == 2:
    print("Usage: key")
    sys.exit(1)

#User provided encryption key via command line argument
key = int(sys.argv[1])

#conditions for key
if key < 1:
    print("Usage: key")
    sys.exit(2)

#Prompt user for text to be encrypted
plaintext = get_string("plaintext: ")

print("ciphertext: ", end="")


for digit in plaintext:
    #Checks for non alphabetic input and outputs without changes
    if not digit.isalpha():
        print(digit, end="")
        continue

    #Takes alpabetic input, preserves case, encrypts, and outputs ciphertext
    ascii_offset = 65 if digit.isupper() else 97
    index = ord(digit) - ascii_offset
    encrypt = (index + key) % 26

    print(chr(encrypt + ascii_offset), end="")

print()

File: pset2/caesar/caesar.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<stdlib.h>
#include<ctype.h>

//Takes plaintext and key from user and as input, and outputs ciphertext
int main(int argc, string argv[])
{
    //Conditions for command line argument
    if (argc != 2)
    {
        printf("Usage: key\n");
        return 1;
    }

    //argument 1 is program name and argument 2 is cryptography key
    int key = atoi(argv[1]);

    if (key <= 0)
    {
        printf("Usage: key\n");
        return 1;
    }

    //Prompt the user for text that needs to be encrypted
    string plaintext = get_string("plaintext: ");

    printf("ciphertext: ");

    //Uses the key supplied by user to encrypt plaintext
    for (int i = 0, n = strlen(plaintext); i < n; i++)
    {
        if (isalpha(plaintext[i]))
        {
            if (isupper(plaintext[i]))
            {
                //takes uppercase plaintext as input, preserves case, and outputs uppercase ciphertext
                printf("%c", (plaintext[i] + key - 'A') % 26 + 'A');
            }
            else if (islower(plaintext[i]))
            {
                //takes lowercase plaintext as input, preserves case, and outputs lowercase ciphertext
                printf("%c", (plaintext[i] + key - 'a') % 26 + 'a');
            }
        }
        else
        {
            //takes non-alphabetic input and outputs without changes
            printf("%c", plaintext[i]);
        }

    }
    printf("\n");
}